File: python/字符串/383. 赎金信.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        import collections
        d = dict(collections.Counter(magazine))
        for i in ransomNote:
            if i not in ransomNote:
                return False
            if d[i]<0:
                return False
            d[i]-=1
        return True
s = Solution()
print(s.canConstruct("aa", "aab"))

File: go/string/383. 赎金信.go
package string

func canConstruct(ransomNote string, magazine string) bool {
	d := make(map[rune]int)
	for _, v := range magazine {
		d[v]++
	}
	for _, v := range ransomNote {
		if _, ok := d[v]; ok {
			d[v]--
			if d[v] < 0 {
				return false
			}
		} else {
			return false
		}
	}
	return true
}
